src/plugins/inspector/public/views/data/components/download_options.tsx

```diff
@@ -0,0 +1,166 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { EuiButton, EuiContextMenuItem, EuiContextMenuPanel, EuiPopover } from '@elastic/eui';
+import { DataViewColumn, DataViewRow } from '../types';
+import { UiSettingsClientContract } from '../../../../../../core/public';
+
+import { exportAsCsv } from '../lib/export_csv';
+
+interface DataDownloadOptionsState {
+  isPopoverOpen: boolean;
+}
+
+interface DataDownloadOptionsProps {
+  title: string;
+  isFormatted?: boolean;
+  columns: DataViewColumn[];
+  rows: DataViewRow[];
+  uiSettings: UiSettingsClientContract;
+}
+
+class DataDownloadOptions extends Component<DataDownloadOptionsProps, DataDownloadOptionsState> {
+  static propTypes = {
+    title: PropTypes.string.isRequired,
+    isFormatted: PropTypes.bool,
+    columns: PropTypes.array,
+    rows: PropTypes.array,
+    uiSettings: PropTypes.object.isRequired,
+  };
+
+  state = {
+    isPopoverOpen: false,
+  };
+
+  csvSeparator = this.props.uiSettings.get('csv:separator', ',');
+  quoteValues = this.props.uiSettings.get('csv:quoteValues', true);
```

Comment:
If `uiSettings` service is needed only for these settings, maybe we could pass these as props.

```tsx
<DataDownloadOptions csvSeparator={...} quoteValues={...} />
```

Reply:
Make sense, thanks